### PR TITLE
103 show reward token of cerc20pluginrewardsdelegate assets

### PR DESCRIPTION
**Pull request by peetzweg**
## Description

Refactors asset deployment, so that it will deploy a new Flywheel with Dynamic Rewards. We weren't able to use our predeployed flywheel because the admin of the pool needed to call 
`FlywheelCore.addStrategyForRewards`, which can only be done by an Authorized signer.

https://github.com/fei-protocol/flywheel-v2/blob/5ac5d314362188e3ece31a9db7deba529f4e8cc7/src/FlywheelCore.sol#L141-L144

Screenshot 2022-06-09 at 14 56 36

Not the prettiest way to doing it, yet again clicking through a bunch of transaction calls.

@carlomazzaferro  can you confirm these cyclelength here:
https://github.com/Midas-Protocol/monorepo/blob/439d31c262669b5efe3dcca24ff51ef274a5ed40/packages/sdk/src/chainConfig/plugin.ts#L111-L122

Just copied them from the bsc.ts deployment. We can also remove the code for deploying our pre deployed flywheels for the strategies.

### update

I get a bad feeling. Not sure what to remove from the deployemnts. But I guess the strategy does need the flywheel as well, so we maybe need the pool admin to deploy his own Strategy as well?!

https://github.com/Midas-Protocol/monorepo/blob/439d31c262669b5efe3dcca24ff51ef274a5ed40/packages/sdk/chainDeploy/helpers/erc4626Plugins.ts#L62-L92

https://github.com/Midas-Protocol/contracts/blob/fc2050f041c675cc7cff3240bcd6dbcb22a4a0b6/contracts/compound/strategies/DotDotLpERC4626.sol#L55-L67

### CODE REVIEW

File: packages/sdk/src/chainConfig/plugin.ts [hunk 1]
```diff
@@ -91,14 +89,12 @@ const chainPluginConfig: ChainPlugins = {
...
-            // FuseFlywheelCore_EPX.json
-            address: "0xC6431455AeE17a08D6409BdFB18c4bc73a4069E4",
+            rewardsCycleLength: 1, // TODO is this correct?
             // EPX rewards
             rewardToken: "0xaf41054c1487b0e5e2b9250c0332ecbce6ce9d71",
           },
           {
-            // FuseFlywheelCore_DDD.json
-            address: "0x851Cc0037B6923e60dC81Fa79Ac0799cC983492c",
+            rewardsCycleLength: 1, // TODO is this correct?
```
Comment (carlomazzaferro) [inline packages/sdk/src/chainConfig/plugin.ts:97]: @RedVeil was the one who initially set these, can't recall what the heuristics were

Comment (carlomazzaferro) [inline packages/sdk/src/chainConfig/plugin.ts:97]: @RedVeil was the one who initially set these, can't recall what the heuristics were

Comment (RedVeil) [reply packages/sdk/src/chainConfig/plugin.ts]: as stated above

File: packages/sdk/src/modules/Asset.ts [hunk 2]
```diff
@@ -162,6 +167,8 @@ export function withAsset<TBase extends FuseBaseConstructor>(Base: TBase) {
             if (addRdTxReceipt.status != constants.One.toNumber()) {
               throw `Failed set add RD to pool ${flywheelConfig.address}`;
             }
+
+            //2. Approve Flywheel to spend underlying
```
Comment (carlomazzaferro) [inline packages/sdk/src/modules/Asset.ts:171]: amazing documentation work!

File: packages/sdk/src/modules/CreateContracts.ts [hunk 3]
```diff
@@ -7,21 +7,17 @@ import { FuseFlywheelCore } from "../../lib/contracts/typechain/FuseFlywheelCore
...
 
 export function withCreateContracts<TBase extends FuseBaseConstructor>(Base: TBase) {
   return class CreateContracts extends Base {
-    createComptroller(comptrollerAddress: string) {
-      return new Contract(
-        comptrollerAddress,
-        this.chainDeployment.Comptroller.abi,
-        this.provider.getSigner()
-      ) as Comptroller;
+    createContractInstance<T extends Contract>(contract: keyof Artifacts) {
```
Comment (carlomazzaferro) [inline packages/sdk/src/modules/CreateContracts.ts:14]: this is refreshingly _cute_

File: packages/ui/hooks/useFuseContract.ts [hunk 4]
```diff
@@ -1,36 +1,36 @@
-import { Comptroller } from '@midas-capital/sdk/dist/cjs/lib/contracts/typechain/Comptroller';
-import { FlywheelStaticRewards } from '@midas-capital/sdk/dist/cjs/lib/contracts/typechain/FlywheelStaticRewards';
-import { FuseFlywheelCore } from '@midas-capital/sdk/dist/cjs/lib/contracts/typechain/FuseFlywheelCore';
-import { RewardsDistributorDelegate } from '@midas-capital/sdk/dist/cjs/lib/contracts/typechain/RewardsDistributorDelegate';
-import { useContract } from 'wagmi';
+// import { Comptroller } from '@midas-capital/sdk/dist/cjs/lib/contracts/typechain/Comptroller';
```
Comment (carlomazzaferro) [inline packages/ui/hooks/useFuseContract.ts:1]: remove this

**Review by carlomazzaferro — APPROVED** [review]
Overall, really like these! Just a few comments here and here

File: packages/sdk/tests/unit/FlywheelModule.spec.ts [hunk 5]
```diff
@@ -127,75 +127,4 @@ describe("FlywheelModule", function () {
 
     // TODO test claimable functions
   });
-
-  it("1 Pool, 1 Flywheel, 1 Reward Distributor", async function () {
```
[the remark this answers is not in the document]
Comment (peetzweg) [reply packages/sdk/tests/unit/FlywheelModule.spec.ts:131]: Flywheel and Rewards Distributors are not supposed to work anymore and we are replacing the rewards distributor completely with flywheel + borrow booster. 
So this code is just dead and won't probably be used ever again.

**Comment by peetzweg** [comment]
@carlomazzaferro i think despite failing ci this pr could be merged. It's seems like the current CI hiccups we have. Nothing criticial.

**Pull request merged**